### DISCUSSION TITLE
panic: Positions [(2931, BeforeSticky)] are out of range for changeset len 2930!

### DISCUSSION THREAD

**Discussion by bcspragu** [helix-editor/helix#11697] [Troubleshooting]
### Summary

I was editing some files and got the following crash on the latest Helix (HEAD):

```
thread 'main' panicked at helix-core/src/transaction.rs:499:9:
Positions [(2931, BeforeSticky)] are out of range for changeset len 2930!
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Reproduction Steps

Unfortunately, I have no useful info here, it crashed abruptly and I don't remember what I was working on. Probably some combination of Go, SQL, and TypeScript files.

I'll turn on RUST_BACKTRACE at a minimum in case it happens again.

### Helix log

I checked the log, but not until much later, couldn't see any logs related to the crash

### Platform

Linux

### Terminal Emulator

alacritty 0.13.2

### Installation Method

source, via Arch Linux AUR

### Helix Version

helix 24.7 (237cbe4b)

**Comment by the-mikedavis** [comment]
Without a reproduction case it's very hard to debug this panic in particular. If you can find a reproduction case please add a comment. It might also be helpful to see the backtrace (but that would only tell us the general area that caused this - we might still need a reproduction case)
